control-plane-operator/controllers/hostedcontrolplane/v2/ocm/config.go

```diff
@@ -96,9 +96,11 @@ func adaptConfig(cfg *openshiftcpv1.OpenShiftControllerManagerConfig, configurat
 		cfg.Build.BuildOverrides.Tolerations = buildConfig.Spec.BuildOverrides.Tolerations
 	}
 
-	// network config
-	if cidrs := configuration.GetAutoAssignCIDRs(); len(cidrs) > 0 {
-		cfg.Ingress.IngressIPNetworkCIDR = cidrs[0]
+	if capabilities.IsIngressCapabilityEnabled(caps) {
+		// network config
```

Comment:
This is for cloud provider interaction with services of type LoadBalancer, I'm not sure we want to disable it. If we want to we should probably add a CEL rule to prevent you from setting this if ingress is disabled.